### PR TITLE
Feat: commission document management

### PR DESCRIPTION
**Pull request by kmdavidds**
Database Migrations:
- Added migration files
- Implemented Makefile commands for database up/down
- Fixed migration down-to-0 functionality

Development Environment:
- Set up development database using Docker Compose
- Added Makefile for migration commands
- Added .gitignore and .env.example

MinIO Integration:
- Updated .env.example with MinIO credentials
- Updated configuration to include MinIO settings

Document Management Features:
- Implemented upload and download functions
- Added upload and download HTTP handlers

### CODE REVIEW

**Comment by mirzahilmi** [comment]
what are the endpoints here that available?

**Comment by kmdavidds** [comment]
The endpoints are:
```
POST /api/v1/commissions/:documentType
GET /api/v1/commissions/:documentType?name=
``` 
The POST endpoint will upload the attached document and the GET endpoint will download a document based on the name of the document

**Comment by mirzahilmi** [comment]
those are the minor reviews, i havent checked if the code actually working or not. I will  check it later, perhaps in the meantime you can provide any test result? @kmdavidds

**Comment by kmdavidds** [comment]
Okay, here are some examples from Postman
POST Endpoint:
image
GET Endpoint:
image

**Comment by mirzahilmi** [comment]
is there any identifier stored to know which document is which? (mou, bast, etc)

**Comment by kmdavidds** [comment]
In minio, each document is in a different directory (e.g., "uploads/mou", "uploads/bast", etc.)
In the relational database, each record has a "document_type" attribute (enum).

**Comment by mirzahilmi** [comment]
Great

> In minio, each document is in a different directory (e.g., "uploads/mous", "uploads/basts", etc.) In the relational database, each record has a "document_type" attribute (enum).

great, could you please add changes to resolve my previous review comment? after that we can merge this PR to dev

**Comment by kmdavidds** [comment]
I'm sorry, the review comment isn't visible from my perspective.
Maybe something is wrong with the authorization?

**Comment by mirzahilmi** [comment]
pardon me, i forgot to submit the request

File: commission-service/compose.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - "5555:5432"
+    volumes:
+      - 'pg_data:/var/lib/postgresql/data'
+    environment:
+      - POSTGRES_USER=${DB_USERNAME}
+      - POSTGRES_PASSWORD=${DB_PASSWORD}
+      - POSTGRES_DB=${DB_NAME}
+
+  minio:
+    image: docker.io/bitnami/minio:latest
```
[the remark this answers is not in the document]
Comment (kmdavidds) [reply commission-service/compose.yaml:16]: Okay

File: commission-service/http/commission.go
[the remark this answers is not in the document]
Comment (mirzahilmi) [reply commission-service/http/commission.go:1]: yes theres 2 of those

[the remark this answers is not in the document]
Comment (kmdavidds) [reply commission-service/http/commission.go:1]: I defined them in package types:
```
const (
	MOU        DocumentType = "mou"
	BAST       DocumentType = "bast"
	DP         DocumentType = "dp"
	Settlement DocumentType = "settlement"
)
```
Also, I always check using a private function in package usecase:
```
func validateDocumentType(documentType string) error {
	if documentType != string(types.MOU) &&
		documentType != string(types.BAST) &&
		documentType != string(types.DP) &&
		documentType != string(types.Settlement) {
		return errors.New("invalid document type")
	}
	return nil
}
```
Is that not enough or is there another way I should implement the const?

[the remark this answers is not in the document]
Comment (kmdavidds) [reply commission-service/http/commission.go:1]: Oh, do you mean I should literally put the two strings in a const?
```
const (
	documentTypeParam = "documentType"
	fileNameQuery = "name"
)
```

[the remark this answers is not in the document]
Comment (mirzahilmi) [reply commission-service/http/commission.go:1]: > Oh, do you mean I should literally put the two strings in a const?
> 
> ```
> const (
> 	documentTypeParam = "documentType"
> 	fileNameQuery = "name"
> )
> ```

yes, things that will always be the same

[the remark this answers is not in the document]
Comment (kmdavidds) [reply commission-service/http/commission.go:1]: Okay, I understand it now

**Comment by kmdavidds** [comment]
I've made changes to resolve your previous review comments

**Review by mirzahilmi — APPROVED** [review]
LGTM

**Pull request merged**